Test failure on i386

A very large number of tests fail on i386 architectures (full log can be found [here](https://salsa.debian.org/debian-gis-team/flox/-/jobs/3765665)).
In all cases the issue seems to be related to a data type mismatch.

Is this considered a real issue?

```
============================= test session starts ==============================
platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0+repack
rootdir: /builds/debian-gis-team/flox/debian/output/source_dir, configfile: pyproject.toml
collected 8076 items / 48 deselected / 8028 selected

../../../tests/test_core.py ...ss..........ss..........ss..........ss... [  0%]
.......ss..........ss........FFssFFFFFFFFFFssFFFFFFFFFFssFFFFFFFFFFssFFF [  1%]
FFFFFFFssFFFFFFFFFFssFFFFFFFF..ss..........ss..........ss..........ss... [  2%]
.......ss..........ss................................................... [  3%]
.....................................................sssssssssssssssssss [  4%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [  5%]
sssss................................................................... [  5%]
........................................................................ [  6%]
........................................................................ [  7%]
........................................................................ [  8%]
........................................................................ [  9%]
........................................................................ [ 10%]
........................................................................ [ 11%]
........................................................................ [ 12%]
.....sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 13%]
sssssssssssssssssssssssssssss........................................... [ 14%]
.....................................................sssssssssssssssssss [ 14%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 15%]
sssssFFFFFFssssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssssFFFFFFssssssF [ 16%]
FFFFFssssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssssFFFFFFssssssFFFFFFs [ 17%]
sssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssssFFFFFFssssssFFFFFFssssssF [ 18%]
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssssFFFFFFssssssFFFFFFssssssFFFFFFF [ 19%]
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssssFFFFFFssssssFFFFFFssssssFFFFFFFFFFFFF [ 20%]
FFFFFFFFFFFFFFFFFFFFFFFssssssFFFFFFssssssFFFFFFssssssFFFFFFFFFFFFFFFFFFF [ 21%]
FFFFFFFFFFFFFFFFFssssssFFFFFFssssssFFFFFFssssssFFFFFFFFFFFFFFFFFFFFFFFFF [ 22%]
FFFFFFFFFFFssssssFFFFFFssssssFFFFFFssssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF [ 22%]
FFFFFssssssFFFFFFssssssFFFFFFssssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFs [ 23%]
sssssFFFFFFssssssF..ss............ss..ss..ss............ss..ss..ss...... [ 24%]
......ss..ss..ss............ss..ss..ss............ss..ss..ss............ [ 25%]
ss..ss..ss............ss..ss..ss............ss..ss..ss............ss..ss [ 26%]
..ss............ss..ss..ss............ss..ss.............FFxFFFxFFFxF... [ 27%]
.....ssssssss................................................sssssssssss [ 28%]
sssssssssssss........xxssxxxxxxxxxxxxssxxss..ss............ss..ssxxssxxx [ 29%]
xxxxxxxxxssxxss..ss............ss..ss..ss............ss..ss..ss......... [ 30%]
...ss..ssxxssxxxxxxxxxxxxssxxss..ss............ss..ss........ss......... [ 31%]
...ss..ss......ss............ssssss..ss............ssssss..ss..........s [ 31%]
s..........ss..........ss..........ss..........ss........FFssFFFFFFFFFFs [ 32%]
sFFFFFFFFFFssFFFFFFFFFFssFFFFFFFFFFssFFFFFFFFFFssFFFFFFFF..ss..........s [ 33%]
s..........ss..........ss..........ss..........ss....................... [ 34%]
........................................................................ [ 35%]
.........................................................sssssssssssssss [ 36%]
sssssssssssssssssssssssssssssssss....................................... [ 37%]
........................................................................ [ 38%]
........................................................................ [ 39%]
........................................................................ [ 40%]
........................................................................ [ 40%]
........................................................................ [ 41%]
........................................................................ [ 42%]
........................................................................ [ 43%]
........................................................................ [ 44%]
.........ssssssssssssssssssssssssssssssssssssssssssssssss............... [ 45%]
........................................................................ [ 46%]
.........sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 47%]
sssssssssssssssssssssssssssssssssFFFFFFFFssssFFFFFFFFFFFFFFFFFFFFFFFFFFF [ 48%]
FFFFFFFFFFFssssFFFFFFssssssFFFFFFFFssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF [ 48%]
FFFFFssssFFFFFFssssssFFFFFFFFssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFs [ 49%]
sssFFFFFFssssssFFFFFFFFssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssFFF [ 50%]
FFFssssssFFFFFFFFssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssFFFFFFsss [ 51%]
sssFFFFFFFFssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssFFFFFFssssssFFF [ 52%]
FFFFFssssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssFFFFFFssssssFFFFFFFFs [ 53%]
sssFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssFFFFFFssssssFFFFFFFFssssFFF [ 54%]
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssFFFFFFssssssFFFFFFFFssssFFFFFFFFF [ 55%]
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFssssFFFFFFssssssF..ss............ss..ss..ss [ 56%]
............ss..ss..ss............ss..ss..ss............ss..ss..ss...... [ 57%]
......ss..ss...s.............s..ss...s.............s..ss...s............ [ 57%]
.s..ss...s.............s..ss...s.............s..ss...s.............s..ss [ 58%]
.............FFxFFFxFFFxF........ssssssss............................... [ 59%]
.................ssssssssssssssssssssssss........xxxsxxxxxxxxxxxxxsxxss. [ 60%]
.ss............ss..ssxxxsxxxxxxxxxxxxxsxxss..ss............ss..ss...s... [ 61%]
..........s..ss..ss............ss..ssxxxsxxxxxxxxxxxxxsxxss..ss......... [ 62%]
...ss..ss.........s.............s..ss......ss............ssssss..ss..... [ 63%]
.......ssssssxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 64%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 65%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 66%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 66%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 67%]
xxxxxxxxxxxxxssssssssssssssssssssssssssssssssssssssssssssssssxxxxxxxxxxx [ 68%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 69%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 70%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 71%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 72%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 73%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 74%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 74%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 75%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxsssssssssssssssssssssssssssssssssss [ 76%]
sssssssssssssxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 77%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxsssssssssssssssssssssssssssssssssss [ 78%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssxxxxxxxxxss [ 79%]
sxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxsssxxxxxxssssssxxxxxxxxxsssxxxxx [ 80%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxsssxxxxxxssssssxxxxxxxxxsssxxxxxxxxxxx [ 81%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxsssxxxxxxssssssxxxxxxxxxsssxxxxxxxxxxxxxxxxx [ 82%]
xxxxxxxxxxxxxxxxxxxxxxsssxxxxxxssssssxxxxxxxxxsssxxxxxxxxxxxxxxxxxxxxxxx [ 83%]
xxxxxxxxxxxxxxxxsssxxxxxxssssssxxxxxxxxxsssxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 83%]
xxxxxxxxxxsssxxxxxxssssssxxxxxxxxxsssxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 84%]
xxxxsssxxxxxxssssssxxxxxxxxxsssxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxss [ 85%]
sxxxxxxssssssxxxxxxxxxsssxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxsssxxxxx [ 86%]
xssssssxxxxxxxxxsssxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxsssxxxxxxsssss [ 87%]
sxxxxsxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxs [ 88%]
xxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxsxxxxxx [ 89%]
xxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxx [ 90%]
xsxxssxxxsxxxxxxxxxxxxxsxxssxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxssssxxx [ 91%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxssssxxxxxxxxssssssssxxx [ 92%]
xxxxxxxxsxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssx [ 92%]
xxsxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxsxxx [ 93%]
xxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxssxxxxxxxxxsxxxxxxxxxxxxxsxxssxxxxxxx [ 94%]
sxxxxxxxxxxxxxsxxssxxxsxxxxxxxxxxxxxsxxss........F.........F....XXX..... [ 95%]
.................xxFFxxFF..FFFsFFsFFsFFsFFsFFsFFFFFFFFFFFFFFFFFFFF...FF. [ 96%]
...............................                                          [ 96%]
../../../tests/test_xarray.py .....................ssssss...ssssss.s...s [ 97%]
......................................................ssssss...ssssss.s. [ 98%]
..s.................................xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx [ 99%]
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx................s...ssss          [100%]

=================================== FAILURES ===================================

[CUT]

FAILED ../../../tests/test_core.py::test_multiple_groupers[False-by22-by11-expected_groups1] - AssertionError: a and b have different dtypes: (a: int64, b: int32)
FAILED ../../../tests/test_core.py::test_multiple_groupers[False-by22-by11-expected_groups2] - AssertionError: a and b have different dtypes: (a: int64, b: int32)
FAILED ../../../tests/test_core.py::test_factorize_reindex_sorting_strings - AssertionError: a and b have different dtypes: (a: int32, b: int64)
FAILED ../../../tests/test_core.py::test_factorize_reindex_sorting_ints - AssertionError: a and b have different dtypes: (a: int32, b: int64)
= 1181 failed, 2884 passed, 1641 skipped, 48 deselected, 2319 xfailed, 3 xpassed, 64 warnings in 292.84s (0:04:52) =
```

Unfortunately the build dependencies cannot currently be installed:
```
python3-pandas : Breaks: python3-dask (<= 2022.02.0+dfsg-2) but 2022.02.0+dfsg-2 is to be installed
```
We'll need to build without python3-dask for the time being.

The testsuite still fails with many instances like this:
```
_ test_groupby_reduce[flox-sum-array0-by0-expected0-expected_groups1-False-float] _
../../../tests/test_core.py:157: in test_groupby_reduce
    assert_equal(groups, np.array([0, 1, 2], g_dtype))
E   AssertionError: a and b have different dtypes: (a: int64, b: int32)
```
https://salsa.debian.org/debian-gis-team/flox/-/jobs/3789060/raw

Only a few remaining failures:
```python
=================================== FAILURES ===================================
_____________________ test_numpy_reduce_axis_subset[flox] ______________________
../../../tests/test_core.py:411: in test_numpy_reduce_axis_subset
    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.int64))
E   AssertionError: a and b have different dtypes: (a: int32, b: int64)
_____________________ test_numpy_reduce_axis_subset[numpy] _____________________
../../../tests/test_core.py:411: in test_numpy_reduce_axis_subset
    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.int64))
E   AssertionError: a and b have different dtypes: (a: int32, b: int64)
______________________ test_factorize_values_outside_bins ______________________
../../../tests/test_core.py:971: in test_factorize_values_outside_bins
    assert_equal(expected, actual)
E   AssertionError: a and b have different dtypes: (a: int64, b: int32)
=============================== warnings summary ===============================
```
https://salsa.debian.org/debian-gis-team/flox/-/jobs/3790984/raw

One down, two to go:
```python
=================================== FAILURES ===================================
_____________________ test_numpy_reduce_axis_subset[flox] ______________________
../../../tests/test_core.py:423: in test_numpy_reduce_axis_subset
    assert_equal(result, expected)
E   AssertionError: a and b have different dtypes: (a: int32, b: int64)
_____________________ test_numpy_reduce_axis_subset[numpy] _____________________
../../../tests/test_core.py:423: in test_numpy_reduce_axis_subset
    assert_equal(result, expected)
E   AssertionError: a and b have different dtypes: (a: int32, b: int64)
=============================== warnings summary ===============================
```
https://salsa.debian.org/debian-gis-team/flox/-/jobs/3791565/raw

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -956,6 +956,7 @@ def test_group_by_datetime(engine, method):
 
 def test_factorize_values_outside_bins():
 
+    # pd.factorize returns intp
     vals = factorize_(
         (np.arange(10).reshape(5, 2), np.arange(10).reshape(5, 2)),
         axis=(0, 1),
@@ -967,7 +968,7 @@ def test_factorize_values_outside_bins():
         fastpath=True,
     )
     actual = vals[0]
-    expected = np.array([[-1, -1], [-1, 0], [6, 12], [18, 24], [-1, -1]], np.int64)
+    expected = np.array([[-1, -1], [-1, 0], [6, 12], [18, 24], [-1, -1]], np.intp)
     assert_equal(expected, actual)
 
 

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -140,7 +140,7 @@ def test_groupby_reduce(
     elif func == "sum":
         expected_result = np.array(expected, dtype=dtype)
     elif func == "count":
-        expected_result = np.array(expected, dtype=np.intp)
+        expected_result = np.array(expected, dtype=np.int64)
 
     result, groups, = groupby_reduce(
         array,
@@ -150,9 +150,9 @@ def test_groupby_reduce(
         fill_value=123,
         engine=engine,
     )
-    # we use pd.Index(expected_groups).to_numpy() which is always intp
+    # we use pd.Index(expected_groups).to_numpy() which is always int64
     # for the values in this tests
-    g_dtype = by.dtype if expected_groups is None else np.intp
+    g_dtype = by.dtype if expected_groups is None else np.int64
 
     assert_equal(groups, np.array([0, 1, 2], g_dtype))
     assert_equal(expected_result, result)
@@ -284,7 +284,7 @@ def test_groupby_reduce_count():
     array = np.array([0, 0, np.nan, np.nan, np.nan, 1, 1])
     labels = np.array(["a", "b", "b", "b", "c", "c", "c"])
     result, _ = groupby_reduce(array, labels, func="count")
-    assert_equal(result, np.array([1, 1, 2], dtype=np.intp))
+    assert_equal(result, np.array([1, 1, 2], dtype=np.int64))
 
 
 def test_func_is_aggregation():
@@ -299,7 +299,7 @@ def test_func_is_aggregation():
 
 @requires_dask
 @pytest.mark.parametrize("func", ("sum", "prod"))
-@pytest.mark.parametrize("dtype", [np.float32, np.float64, np.int32, np.intp])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64, np.int32, np.int64])
 def test_groupby_reduce_preserves_dtype(dtype, func):
     array = np.ones((2, 12), dtype=dtype)
     by = np.array([labels] * 2)
@@ -405,32 +405,32 @@ def test_groupby_agg_dask(func, shape, array_chunks, group_chunks, add_nan, dtyp
 def test_numpy_reduce_axis_subset(engine):
     # TODO: add NaNs
     by = labels2d
-    array = np.ones_like(by, dtype=np.intp)
+    array = np.ones_like(by, dtype=np.int64)
     kwargs = dict(func="count", engine=engine, fill_value=0)
     result, _ = groupby_reduce(array, by, **kwargs, axis=1)
-    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.intp))
+    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.int64))
 
     by = np.broadcast_to(labels2d, (3, *labels2d.shape))
     array = np.ones_like(by)
     result, _ = groupby_reduce(array, by, **kwargs, axis=1)
-    subarr = np.array([[1, 1], [1, 1], [0, 2], [1, 1], [1, 1]], dtype=np.intp)
+    subarr = np.array([[1, 1], [1, 1], [0, 2], [1, 1], [1, 1]], dtype=np.int64)
     expected = np.tile(subarr, (3, 1, 1))
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by, **kwargs, axis=2)
-    subarr = np.array([[2, 3], [2, 3]], dtype=np.intp)
+    subarr = np.array([[2, 3], [2, 3]], dtype=np.int64)
     expected = np.tile(subarr, (3, 1, 1))
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by, **kwargs, axis=(1, 2))
-    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.intp)
+    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.int64)
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by, **kwargs, axis=(2, 1))
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by[0, ...], **kwargs, axis=(1, 2))
-    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.intp)
+    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.int64)
     assert_equal(result, expected)
 
 
@@ -438,7 +438,7 @@ def test_numpy_reduce_axis_subset(engine):
 def test_dask_reduce_axis_subset():
 
     by = labels2d
-    array = np.ones_like(by, dtype=np.intp)
+    array = np.ones_like(by, dtype=np.int64)
     with raise_if_dask_computes():
         result, _ = groupby_reduce(
             da.from_array(array, chunks=(2, 3)),
@@ -447,11 +447,11 @@ def test_dask_reduce_axis_subset():
             axis=1,
             expected_groups=[0, 2],
         )
-    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.intp))
+    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.int64))
 
     by = np.broadcast_to(labels2d, (3, *labels2d.shape))
     array = np.ones_like(by)
-    subarr = np.array([[1, 1], [1, 1], [123, 2], [1, 1], [1, 1]], dtype=np.intp)
+    subarr = np.array([[1, 1], [1, 1], [123, 2], [1, 1], [1, 1]], dtype=np.int64)
     expected = np.tile(subarr, (3, 1, 1))
     with raise_if_dask_computes():
         result, _ = groupby_reduce(
@@ -464,7 +464,7 @@ def test_dask_reduce_axis_subset():
         )
     assert_equal(result, expected)
 
-    subarr = np.array([[2, 3], [2, 3]], dtype=np.intp)
+    subarr = np.array([[2, 3], [2, 3]], dtype=np.int64)
     expected = np.tile(subarr, (3, 1, 1))
     with raise_if_dask_computes():
         result, _ = groupby_reduce(
@@ -580,9 +580,9 @@ def test_groupby_all_nan_blocks_dask(expected_groups, reindex, engine):
     nan_labels[:5] = np.nan
 
     array, by, expected = (
-        np.ones((2, 12), dtype=np.intp),
+        np.ones((2, 12), dtype=np.int64),
         np.array([nan_labels, nan_labels[::-1]]),
-        np.array([2, 8, 4], dtype=np.intp),
+        np.array([2, 8, 4], dtype=np.int64),
     )
 
     actual, _ = groupby_reduce(
@@ -672,7 +672,7 @@ def test_groupby_bins(chunk_labels, chunks, engine, method) -> None:
             engine=engine,
             method=method,
         )
-    expected = np.array([3, 1, 0], dtype=np.intp)
+    expected = np.array([3, 1, 0], dtype=np.int64)
     for left, right in zip(groups, pd.IntervalIndex.from_arrays([1, 2, 4], [2, 4, 5]).to_numpy()):
         assert left == right
     assert_equal(actual, expected)
@@ -772,7 +772,7 @@ def test_fill_value_behaviour(func, chunks, fill_value, engine):
 
 @requires_dask
 @pytest.mark.parametrize("func", ["mean", "sum"])
-@pytest.mark.parametrize("dtype", ["float32", "float64", "int32", "intp"])
+@pytest.mark.parametrize("dtype", ["float32", "float64", "int32", "int64"])
 def test_dtype_preservation(dtype, func, engine):
     if func == "sum" or (func == "mean" and "float" in dtype):
         expected = np.dtype(dtype)
@@ -789,8 +789,10 @@ def test_dtype_preservation(dtype, func, engine):
 
 
 @requires_dask
-@pytest.mark.parametrize("dtype", [np.int32, np.intp])
-@pytest.mark.parametrize("labels_dtype", [pytest.param(np.int32, marks=pytest.mark.xfail), np.intp])
+@pytest.mark.parametrize("dtype", [np.int32, np.int64])
+@pytest.mark.parametrize(
+    "labels_dtype", [pytest.param(np.int32, marks=pytest.mark.xfail), np.int64]
+)
 @pytest.mark.parametrize("method", ["map-reduce", "cohorts"])
 def test_cohorts_map_reduce_consistent_dtypes(method, dtype, labels_dtype):
     repeats = np.array([4, 4, 12, 2, 3, 4], dtype=np.int32)
@@ -799,7 +801,7 @@ def test_cohorts_map_reduce_consistent_dtypes(method, dtype, labels_dtype):
 
     actual, actual_groups = groupby_reduce(array, labels, func="count", method=method)
     assert_equal(actual_groups, np.arange(6, dtype=labels.dtype))
-    assert_equal(actual, repeats.astype(np.intp))
+    assert_equal(actual, repeats.astype(np.int64))
 
     actual, actual_groups = groupby_reduce(array, labels, func="sum", method=method)
     assert_equal(actual_groups, np.arange(6, dtype=labels.dtype))
@@ -815,7 +817,7 @@ def test_cohorts_nd_by(func, method, axis, engine):
     o2 = dask.array.ones((2, 3), chunks=-1)
 
     array = dask.array.block([[o, 2 * o], [3 * o2, 4 * o2]])
-    by = array.compute().astype(np.intp)
+    by = array.compute().astype(np.int64)
     by[0, 1] = 30
     by[2, 1] = 40
     by[0, 4] = 31
@@ -840,9 +842,9 @@ def test_cohorts_nd_by(func, method, axis, engine):
 
     actual, groups = groupby_reduce(array, by, sort=False, **kwargs)
     if method == "map-reduce":
-        assert_equal(groups, np.array([1, 30, 2, 31, 3, 4, 40], dtype=np.intp))
+        assert_equal(groups, np.array([1, 30, 2, 31, 3, 4, 40], dtype=np.int64))
     else:
-        assert_equal(groups, np.array([1, 30, 2, 31, 3, 40, 4], dtype=np.intp))
+        assert_equal(groups, np.array([1, 30, 2, 31, 3, 40, 4], dtype=np.int64))
     reindexed = reindex_(actual, groups, pd.Index(sorted_groups))
     assert_equal(reindexed, expected)
 
@@ -965,7 +967,7 @@ def test_factorize_values_outside_bins():
         fastpath=True,
     )
     actual = vals[0]
-    expected = np.array([[-1, -1], [-1, 0], [6, 12], [18, 24], [-1, -1]], np.intp)
+    expected = np.array([[-1, -1], [-1, 0], [6, 12], [18, 24], [-1, -1]], np.int64)
     assert_equal(expected, actual)
 
 
@@ -976,7 +978,7 @@ def test_multiple_groupers_bins(chunk) -> None:
 
     xp = dask.array if chunk else np
     array_kwargs = {"chunks": 2} if chunk else {}
-    array = xp.ones((5, 2), **array_kwargs, dtype=np.intp)
+    array = xp.ones((5, 2), **array_kwargs, dtype=np.int64)
 
     actual, *_ = groupby_reduce(
         array,
@@ -989,7 +991,7 @@ def test_multiple_groupers_bins(chunk) -> None:
         ),
         func="count",
     )
-    expected = np.eye(5, 5, dtype=np.intp)
+    expected = np.eye(5, 5, dtype=np.int64)
     assert_equal(expected, actual)
 
 
@@ -1013,12 +1015,12 @@ def test_multiple_groupers(chunk, by1, by2, expected_groups) -> None:
 
     xp = dask.array if chunk else np
     array_kwargs = {"chunks": 2} if chunk else {}
-    array = xp.ones((5, 2), **array_kwargs, dtype=np.intp)
+    array = xp.ones((5, 2), **array_kwargs, dtype=np.int64)
 
     if chunk:
         by2 = dask.array.from_array(by2)
 
-    expected = np.ones((5, 2), dtype=np.intp)
+    expected = np.ones((5, 2), dtype=np.int64)
     actual, *_ = groupby_reduce(
         array, by1, by2, axis=(0, 1), func="count", expected_groups=expected_groups
     )
@@ -1064,38 +1066,38 @@ def test_factorize_reindex_sorting_strings():
     )
 
     expected = factorize_(**kwargs, reindex=True, sort=True)[0]
-    assert_equal(expected, np.array([0, 1, 4, 2], dtype=np.intp))
+    assert_equal(expected, np.array([0, 1, 4, 2], dtype=np.int64))
 
     expected = factorize_(**kwargs, reindex=True, sort=False)[0]
-    assert_equal(expected, np.array([0, 3, 4, 1], dtype=np.intp))
+    assert_equal(expected, np.array([0, 3, 4, 1], dtype=np.int64))
 
     expected = factorize_(**kwargs, reindex=False, sort=False)[0]
-    assert_equal(expected, np.array([0, 1, 2, 3], dtype=np.intp))
+    assert_equal(expected, np.array([0, 1, 2, 3], dtype=np.int64))
 
     expected = factorize_(**kwargs, reindex=False, sort=True)[0]
-    assert_equal(expected, np.array([0, 1, 3, 2], dtype=np.intp))
+    assert_equal(expected, np.array([0, 1, 3, 2], dtype=np.int64))
 
 
 def test_factorize_reindex_sorting_ints():
     kwargs = dict(
         by=(np.array([-10, 1, 10, 2, 3, 5]),),
         axis=-1,
-        expected_groups=(np.array([0, 1, 2, 3, 4, 5], np.intp),),
+        expected_groups=(np.array([0, 1, 2, 3, 4, 5], np.int64),),
     )
 
     expected = factorize_(**kwargs, reindex=True, sort=True)[0]
-    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.intp))
+    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.int64))
 
     expected = factorize_(**kwargs, reindex=True, sort=False)[0]
-    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.intp))
+    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.int64))
 
     kwargs["expected_groups"] = (np.arange(5, -1, -1),)
 
     expected = factorize_(**kwargs, reindex=True, sort=True)[0]
-    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.intp))
+    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.int64))
 
     expected = factorize_(**kwargs, reindex=True, sort=False)[0]
-    assert_equal(expected, np.array([6, 4, 6, 3, 2, 0], dtype=np.intp))
+    assert_equal(expected, np.array([6, 4, 6, 3, 2, 0], dtype=np.int64))
 
 
 @requires_dask

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -140,7 +140,7 @@ def test_groupby_reduce(
     elif func == "sum":
         expected_result = np.array(expected, dtype=dtype)
     elif func == "count":
-        expected_result = np.array(expected, dtype=np.int64)
+        expected_result = np.array(expected, dtype=np.intp)
 
     result, groups, = groupby_reduce(
         array,
@@ -284,7 +284,7 @@ def test_groupby_reduce_count():
     array = np.array([0, 0, np.nan, np.nan, np.nan, 1, 1])
     labels = np.array(["a", "b", "b", "b", "c", "c", "c"])
     result, _ = groupby_reduce(array, labels, func="count")
-    assert_equal(result, np.array([1, 1, 2], dtype=np.int64))
+    assert_equal(result, np.array([1, 1, 2], dtype=np.intp))
 
 
 def test_func_is_aggregation():
@@ -413,7 +413,7 @@ def test_numpy_reduce_axis_subset(engine):
     by = np.broadcast_to(labels2d, (3, *labels2d.shape))
     array = np.ones_like(by)
     result, _ = groupby_reduce(array, by, **kwargs, axis=1)
-    subarr = np.array([[1, 1], [1, 1], [0, 2], [1, 1], [1, 1]], dtype=np.int64)
+    subarr = np.array([[1, 1], [1, 1], [0, 2], [1, 1], [1, 1]], dtype=np.intp)
     expected = np.tile(subarr, (3, 1, 1))
     assert_equal(result, expected)
 
@@ -672,7 +672,7 @@ def test_groupby_bins(chunk_labels, chunks, engine, method) -> None:
             engine=engine,
             method=method,
         )
-    expected = np.array([3, 1, 0], dtype=np.int64)
+    expected = np.array([3, 1, 0], dtype=np.intp)
     for left, right in zip(groups, pd.IntervalIndex.from_arrays([1, 2, 4], [2, 4, 5]).to_numpy()):
         assert left == right
     assert_equal(actual, expected)
@@ -991,7 +991,8 @@ def test_multiple_groupers_bins(chunk) -> None:
         ),
         func="count",
     )
-    expected = np.eye(5, 5, dtype=np.int64)
+    # output from `count` is intp
+    expected = np.eye(5, 5, dtype=np.intp)
     assert_equal(expected, actual)
 
 
@@ -1020,7 +1021,8 @@ def test_multiple_groupers(chunk, by1, by2, expected_groups) -> None:
     if chunk:
         by2 = dask.array.from_array(by2)
 
-    expected = np.ones((5, 2), dtype=np.int64)
+    # output from `count` is intp
+    expected = np.ones((5, 2), dtype=np.intp)
     actual, *_ = groupby_reduce(
         array, by1, by2, axis=(0, 1), func="count", expected_groups=expected_groups
     )
@@ -1059,6 +1061,7 @@ def test_validate_expected_groups_not_none_dask() -> None:
 
 
 def test_factorize_reindex_sorting_strings():
+    # pd.factorize seems to return intp so int32 on 32bit arch
     kwargs = dict(
         by=(np.array(["El-Nino", "La-Nina", "boo", "Neutral"]),),
         axis=-1,
@@ -1066,19 +1069,20 @@ def test_factorize_reindex_sorting_strings():
     )
 
     expected = factorize_(**kwargs, reindex=True, sort=True)[0]
-    assert_equal(expected, np.array([0, 1, 4, 2], dtype=np.int64))
+    assert_equal(expected, np.array([0, 1, 4, 2], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=True, sort=False)[0]
-    assert_equal(expected, np.array([0, 3, 4, 1], dtype=np.int64))
+    assert_equal(expected, np.array([0, 3, 4, 1], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=False, sort=False)[0]
-    assert_equal(expected, np.array([0, 1, 2, 3], dtype=np.int64))
+    assert_equal(expected, np.array([0, 1, 2, 3], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=False, sort=True)[0]
-    assert_equal(expected, np.array([0, 1, 3, 2], dtype=np.int64))
+    assert_equal(expected, np.array([0, 1, 3, 2], dtype=np.intp))
 
 
 def test_factorize_reindex_sorting_ints():
+    # pd.factorize seems to return intp so int32 on 32bit arch
     kwargs = dict(
         by=(np.array([-10, 1, 10, 2, 3, 5]),),
         axis=-1,
@@ -1086,18 +1090,18 @@ def test_factorize_reindex_sorting_ints():
     )
 
     expected = factorize_(**kwargs, reindex=True, sort=True)[0]
-    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.int64))
+    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=True, sort=False)[0]
-    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.int64))
+    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.intp))
 
     kwargs["expected_groups"] = (np.arange(5, -1, -1),)
 
     expected = factorize_(**kwargs, reindex=True, sort=True)[0]
-    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.int64))
+    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=True, sort=False)[0]
-    assert_equal(expected, np.array([6, 4, 6, 3, 2, 0], dtype=np.int64))
+    assert_equal(expected, np.array([6, 4, 6, 3, 2, 0], dtype=np.intp))
 
 
 @requires_dask

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -150,9 +150,9 @@ def test_groupby_reduce(
         fill_value=123,
         engine=engine,
     )
-    # we use pd.Index(expected_groups).to_numpy() which is always int64
+    # we use pd.Index(expected_groups).to_numpy() which is always intp
     # for the values in this tests
-    g_dtype = by.dtype if expected_groups is None else np.int64
+    g_dtype = by.dtype if expected_groups is None else np.intp
 
     assert_equal(groups, np.array([0, 1, 2], g_dtype))
     assert_equal(expected_result, result)

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -140,7 +140,7 @@ def test_groupby_reduce(
     elif func == "sum":
         expected_result = np.array(expected, dtype=dtype)
     elif func == "count":
-        expected_result = np.array(expected, dtype=np.int64)
+        expected_result = np.array(expected, dtype=np.intp)
 
     result, groups, = groupby_reduce(
         array,
@@ -150,9 +150,9 @@ def test_groupby_reduce(
         fill_value=123,
         engine=engine,
     )
-    # we use pd.Index(expected_groups).to_numpy() which is always int64
+    # we use pd.Index(expected_groups).to_numpy() which is always intp
     # for the values in this tests
-    g_dtype = by.dtype if expected_groups is None else np.int64
+    g_dtype = by.dtype if expected_groups is None else np.intp
 
     assert_equal(groups, np.array([0, 1, 2], g_dtype))
     assert_equal(expected_result, result)
@@ -284,7 +284,7 @@ def test_groupby_reduce_count():
     array = np.array([0, 0, np.nan, np.nan, np.nan, 1, 1])
     labels = np.array(["a", "b", "b", "b", "c", "c", "c"])
     result, _ = groupby_reduce(array, labels, func="count")
-    assert_equal(result, np.array([1, 1, 2], dtype=np.int64))
+    assert_equal(result, np.array([1, 1, 2], dtype=np.intp))
 
 
 def test_func_is_aggregation():
@@ -299,7 +299,7 @@ def test_func_is_aggregation():
 
 @requires_dask
 @pytest.mark.parametrize("func", ("sum", "prod"))
-@pytest.mark.parametrize("dtype", [np.float32, np.float64, np.int32, np.int64])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64, np.int32, np.intp])
 def test_groupby_reduce_preserves_dtype(dtype, func):
     array = np.ones((2, 12), dtype=dtype)
     by = np.array([labels] * 2)
@@ -405,32 +405,32 @@ def test_groupby_agg_dask(func, shape, array_chunks, group_chunks, add_nan, dtyp
 def test_numpy_reduce_axis_subset(engine):
     # TODO: add NaNs
     by = labels2d
-    array = np.ones_like(by, dtype=np.int64)
+    array = np.ones_like(by, dtype=np.intp)
     kwargs = dict(func="count", engine=engine, fill_value=0)
     result, _ = groupby_reduce(array, by, **kwargs, axis=1)
-    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.int64))
+    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.intp))
 
     by = np.broadcast_to(labels2d, (3, *labels2d.shape))
     array = np.ones_like(by)
     result, _ = groupby_reduce(array, by, **kwargs, axis=1)
-    subarr = np.array([[1, 1], [1, 1], [0, 2], [1, 1], [1, 1]], dtype=np.int64)
+    subarr = np.array([[1, 1], [1, 1], [0, 2], [1, 1], [1, 1]], dtype=np.intp)
     expected = np.tile(subarr, (3, 1, 1))
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by, **kwargs, axis=2)
-    subarr = np.array([[2, 3], [2, 3]], dtype=np.int64)
+    subarr = np.array([[2, 3], [2, 3]], dtype=np.intp)
     expected = np.tile(subarr, (3, 1, 1))
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by, **kwargs, axis=(1, 2))
-    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.int64)
+    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.intp)
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by, **kwargs, axis=(2, 1))
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by[0, ...], **kwargs, axis=(1, 2))
-    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.int64)
+    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.intp)
     assert_equal(result, expected)
 
 
@@ -438,7 +438,7 @@ def test_numpy_reduce_axis_subset(engine):
 def test_dask_reduce_axis_subset():
 
     by = labels2d
-    array = np.ones_like(by, dtype=np.int64)
+    array = np.ones_like(by, dtype=np.intp)
     with raise_if_dask_computes():
         result, _ = groupby_reduce(
             da.from_array(array, chunks=(2, 3)),
@@ -447,11 +447,11 @@ def test_dask_reduce_axis_subset():
             axis=1,
             expected_groups=[0, 2],
         )
-    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.int64))
+    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.intp))
 
     by = np.broadcast_to(labels2d, (3, *labels2d.shape))
     array = np.ones_like(by)
-    subarr = np.array([[1, 1], [1, 1], [123, 2], [1, 1], [1, 1]], dtype=np.int64)
+    subarr = np.array([[1, 1], [1, 1], [123, 2], [1, 1], [1, 1]], dtype=np.intp)
     expected = np.tile(subarr, (3, 1, 1))
     with raise_if_dask_computes():
         result, _ = groupby_reduce(
@@ -464,7 +464,7 @@ def test_dask_reduce_axis_subset():
         )
     assert_equal(result, expected)
 
-    subarr = np.array([[2, 3], [2, 3]], dtype=np.int64)
+    subarr = np.array([[2, 3], [2, 3]], dtype=np.intp)
     expected = np.tile(subarr, (3, 1, 1))
     with raise_if_dask_computes():
         result, _ = groupby_reduce(
@@ -580,9 +580,9 @@ def test_groupby_all_nan_blocks_dask(expected_groups, reindex, engine):
     nan_labels[:5] = np.nan
 
     array, by, expected = (
-        np.ones((2, 12), dtype=np.int64),
+        np.ones((2, 12), dtype=np.intp),
         np.array([nan_labels, nan_labels[::-1]]),
-        np.array([2, 8, 4], dtype=np.int64),
+        np.array([2, 8, 4], dtype=np.intp),
     )
 
     actual, _ = groupby_reduce(
@@ -672,7 +672,7 @@ def test_groupby_bins(chunk_labels, chunks, engine, method) -> None:
             engine=engine,
             method=method,
         )
-    expected = np.array([3, 1, 0], dtype=np.int64)
+    expected = np.array([3, 1, 0], dtype=np.intp)
     for left, right in zip(groups, pd.IntervalIndex.from_arrays([1, 2, 4], [2, 4, 5]).to_numpy()):
         assert left == right
     assert_equal(actual, expected)
@@ -772,7 +772,7 @@ def test_fill_value_behaviour(func, chunks, fill_value, engine):
 
 @requires_dask
 @pytest.mark.parametrize("func", ["mean", "sum"])
-@pytest.mark.parametrize("dtype", ["float32", "float64", "int32", "int64"])
+@pytest.mark.parametrize("dtype", ["float32", "float64", "int32", "intp"])
 def test_dtype_preservation(dtype, func, engine):
     if func == "sum" or (func == "mean" and "float" in dtype):
         expected = np.dtype(dtype)
@@ -789,10 +789,8 @@ def test_dtype_preservation(dtype, func, engine):
 
 
 @requires_dask
-@pytest.mark.parametrize("dtype", [np.int32, np.int64])
-@pytest.mark.parametrize(
-    "labels_dtype", [pytest.param(np.int32, marks=pytest.mark.xfail), np.int64]
-)
+@pytest.mark.parametrize("dtype", [np.int32, np.intp])
+@pytest.mark.parametrize("labels_dtype", [pytest.param(np.int32, marks=pytest.mark.xfail), np.intp])
 @pytest.mark.parametrize("method", ["map-reduce", "cohorts"])
 def test_cohorts_map_reduce_consistent_dtypes(method, dtype, labels_dtype):
     repeats = np.array([4, 4, 12, 2, 3, 4], dtype=np.int32)
@@ -801,7 +799,7 @@ def test_cohorts_map_reduce_consistent_dtypes(method, dtype, labels_dtype):
 
     actual, actual_groups = groupby_reduce(array, labels, func="count", method=method)
     assert_equal(actual_groups, np.arange(6, dtype=labels.dtype))
-    assert_equal(actual, repeats.astype(np.int64))
+    assert_equal(actual, repeats.astype(np.intp))
 
     actual, actual_groups = groupby_reduce(array, labels, func="sum", method=method)
     assert_equal(actual_groups, np.arange(6, dtype=labels.dtype))
@@ -817,7 +815,7 @@ def test_cohorts_nd_by(func, method, axis, engine):
     o2 = dask.array.ones((2, 3), chunks=-1)
 
     array = dask.array.block([[o, 2 * o], [3 * o2, 4 * o2]])
-    by = array.compute().astype(np.int64)
+    by = array.compute().astype(np.intp)
     by[0, 1] = 30
     by[2, 1] = 40
     by[0, 4] = 31
@@ -842,9 +840,9 @@ def test_cohorts_nd_by(func, method, axis, engine):
 
     actual, groups = groupby_reduce(array, by, sort=False, **kwargs)
     if method == "map-reduce":
-        assert_equal(groups, np.array([1, 30, 2, 31, 3, 4, 40], dtype=np.int64))
+        assert_equal(groups, np.array([1, 30, 2, 31, 3, 4, 40], dtype=np.intp))
     else:
-        assert_equal(groups, np.array([1, 30, 2, 31, 3, 40, 4], dtype=np.int64))
+        assert_equal(groups, np.array([1, 30, 2, 31, 3, 40, 4], dtype=np.intp))
     reindexed = reindex_(actual, groups, pd.Index(sorted_groups))
     assert_equal(reindexed, expected)
 
@@ -967,7 +965,7 @@ def test_factorize_values_outside_bins():
         fastpath=True,
     )
     actual = vals[0]
-    expected = np.array([[-1, -1], [-1, 0], [6, 12], [18, 24], [-1, -1]], np.int64)
+    expected = np.array([[-1, -1], [-1, 0], [6, 12], [18, 24], [-1, -1]], np.intp)
     assert_equal(expected, actual)
 
 
@@ -978,7 +976,7 @@ def test_multiple_groupers_bins(chunk) -> None:
 
     xp = dask.array if chunk else np
     array_kwargs = {"chunks": 2} if chunk else {}
-    array = xp.ones((5, 2), **array_kwargs, dtype=np.int64)
+    array = xp.ones((5, 2), **array_kwargs, dtype=np.intp)
 
     actual, *_ = groupby_reduce(
         array,
@@ -991,7 +989,7 @@ def test_multiple_groupers_bins(chunk) -> None:
         ),
         func="count",
     )
-    expected = np.eye(5, 5, dtype=np.int64)
+    expected = np.eye(5, 5, dtype=np.intp)
     assert_equal(expected, actual)
 
 
@@ -1015,12 +1013,12 @@ def test_multiple_groupers(chunk, by1, by2, expected_groups) -> None:
 
     xp = dask.array if chunk else np
     array_kwargs = {"chunks": 2} if chunk else {}
-    array = xp.ones((5, 2), **array_kwargs, dtype=np.int64)
+    array = xp.ones((5, 2), **array_kwargs, dtype=np.intp)
 
     if chunk:
         by2 = dask.array.from_array(by2)
 
-    expected = np.ones((5, 2), dtype=np.int64)
+    expected = np.ones((5, 2), dtype=np.intp)
     actual, *_ = groupby_reduce(
         array, by1, by2, axis=(0, 1), func="count", expected_groups=expected_groups
     )
@@ -1066,38 +1064,38 @@ def test_factorize_reindex_sorting_strings():
     )
 
     expected = factorize_(**kwargs, reindex=True, sort=True)[0]
-    assert_equal(expected, np.array([0, 1, 4, 2], dtype=np.int64))
+    assert_equal(expected, np.array([0, 1, 4, 2], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=True, sort=False)[0]
-    assert_equal(expected, np.array([0, 3, 4, 1], dtype=np.int64))
+    assert_equal(expected, np.array([0, 3, 4, 1], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=False, sort=False)[0]
-    assert_equal(expected, np.array([0, 1, 2, 3], dtype=np.int64))
+    assert_equal(expected, np.array([0, 1, 2, 3], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=False, sort=True)[0]
-    assert_equal(expected, np.array([0, 1, 3, 2], dtype=np.int64))
+    assert_equal(expected, np.array([0, 1, 3, 2], dtype=np.intp))
 
 
 def test_factorize_reindex_sorting_ints():
     kwargs = dict(
         by=(np.array([-10, 1, 10, 2, 3, 5]),),
         axis=-1,
-        expected_groups=(np.array([0, 1, 2, 3, 4, 5], np.int64),),
+        expected_groups=(np.array([0, 1, 2, 3, 4, 5], np.intp),),
     )
 
     expected = factorize_(**kwargs, reindex=True, sort=True)[0]
-    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.int64))
+    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=True, sort=False)[0]
-    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.int64))
+    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.intp))
 
     kwargs["expected_groups"] = (np.arange(5, -1, -1),)
 
     expected = factorize_(**kwargs, reindex=True, sort=True)[0]
-    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.int64))
+    assert_equal(expected, np.array([6, 1, 6, 2, 3, 5], dtype=np.intp))
 
     expected = factorize_(**kwargs, reindex=True, sort=False)[0]
-    assert_equal(expected, np.array([6, 4, 6, 3, 2, 0], dtype=np.int64))
+    assert_equal(expected, np.array([6, 4, 6, 3, 2, 0], dtype=np.intp))
 
 
 @requires_dask

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -150,9 +150,9 @@ def test_groupby_reduce(
         fill_value=123,
         engine=engine,
     )
-    # we use pd.Index(expected_groups).to_numpy() which is always intp
+    # we use pd.Index(expected_groups).to_numpy() which is always int64
     # for the values in this tests
-    g_dtype = by.dtype if expected_groups is None else np.intp
+    g_dtype = by.dtype if expected_groups is None else np.int64
 
     assert_equal(groups, np.array([0, 1, 2], g_dtype))
     assert_equal(expected_result, result)

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -418,19 +418,19 @@ def test_numpy_reduce_axis_subset(engine):
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by, **kwargs, axis=2)
-    subarr = np.array([[2, 3], [2, 3]], dtype=np.int64)
+    subarr = np.array([[2, 3], [2, 3]], dtype=np.intp)
     expected = np.tile(subarr, (3, 1, 1))
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by, **kwargs, axis=(1, 2))
-    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.int64)
+    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.intp)
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by, **kwargs, axis=(2, 1))
     assert_equal(result, expected)
 
     result, _ = groupby_reduce(array, by[0, ...], **kwargs, axis=(1, 2))
-    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.int64)
+    expected = np.array([[4, 6], [4, 6], [4, 6]], dtype=np.intp)
     assert_equal(result, expected)
 
 

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -408,7 +408,7 @@ def test_numpy_reduce_axis_subset(engine):
     array = np.ones_like(by, dtype=np.int64)
     kwargs = dict(func="count", engine=engine, fill_value=0)
     result, _ = groupby_reduce(array, by, **kwargs, axis=1)
-    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.int64))
+    assert_equal(result, np.array([[2, 3], [2, 3]], dtype=np.intp))
 
     by = np.broadcast_to(labels2d, (3, *labels2d.shape))
     array = np.ones_like(by)
@@ -955,7 +955,6 @@ def test_group_by_datetime(engine, method):
 
 
 def test_factorize_values_outside_bins():
-
     # pd.factorize returns intp
     vals = factorize_(
         (np.arange(10).reshape(5, 2), np.arange(10).reshape(5, 2)),

